e2e/analysis/analysis_test.go

```diff
@@ -15,123 +15,139 @@
 package analysis_test
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/ZupIT/horusec-devkit/pkg/enums/languages"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
 
+	"github.com/google/uuid"
 	. "github.com/onsi/ginkgo"
+	ginkgoconfig "github.com/onsi/ginkgo/config"
 	. "github.com/onsi/gomega"
-	"github.com/onsi/gomega/gexec"
 
-	"github.com/ZupIT/horusec-devkit/pkg/enums/languages"
-	"github.com/ZupIT/horusec-devkit/pkg/enums/tools"
+	"github.com/ZupIT/horusec-devkit/pkg/utils/logger"
+	"github.com/ZupIT/horusec/config"
+	"github.com/ZupIT/horusec/e2e/analysis"
+	customimages "github.com/ZupIT/horusec/internal/entities/custom_images"
 	"github.com/ZupIT/horusec/internal/utils/testutil"
 )
 
-type AnalysisTestCase struct {
-	Language       languages.Language
-	AnalysisFolder string
-	ExpectedTools  []tools.Tool
-}
+const isWindows = runtime.GOOS == "windows"
+const isDarwin = runtime.GOOS == "darwin"
+const horusecConfigName = "horusec-config.json"
+
+var _ = Describe("Run a complete horusec analysis when build tools locally", func() {
+	tmpDir := CreateHorusecConfigAndReturnTMPDirectory()
+
+	Describe("Running e2e tests when build tools locally", func() {
+		allTestsCases := analysis.NewTestCase()
+
+		for idx := range allTestsCases {
+			if !isDarwin {
+				horusecConfigFilePathBuildLocally := filepath.Join(tmpDir, horusecConfigName)
+				RunTestCase(allTestsCases, horusecConfigFilePathBuildLocally, "(build locally)", idx)
+
+				time.Sleep(2 * time.Second)
+			}
 
-func (A AnalysisTestCase) RunAnalysisTestCase() (*gexec.Session, error) {
-	flags := map[string]string{
-		testutil.StartFlagProjectPath: A.AnalysisFolder,
+			horusecConfigFilePathDownloadFromDockerHub := filepath.Join(allTestsCases[idx].Command.Flags[testutil.StartFlagProjectPath], horusecConfigName)
+			RunTestCase(allTestsCases, horusecConfigFilePathDownloadFromDockerHub, "(download from dockerhub)", idx)
+
+			time.Sleep(2 * time.Second)
+
+			if CurrentGinkgoTestDescription().Failed && ginkgoconfig.GinkgoConfig.FailFast {
+				logger.LogInfo("TestCase executed with: error")
+				Fail("Failed test for tool " + allTestsCases[idx].Tool.ToString())
+				break
+			}
+		}
+	})
+
+	AfterEach(func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			Fail(fmt.Sprintf("Error on remove tmp folder: %v", err))
+		}
+	})
+})
+
+func RunTestCase(allTestsCases []analysis.TestCase, horusecConfigFilePath, runType string, idx int) {
+	testCase := allTestsCases[idx]
```

Comment:
how is necessary set value in `testCase.Command.Output = string(session.Out.Contents())` update value from params is bad pratics for me.

Reply:
But you already changing the []analysis.TestCase. You can pass a copy of analysis.TestCase instead a pointer.

Reply:
in this case is not pointer, because i don't want change values inside of a loop

Reply:
I also think that with the pointer it will be better.

```
func RunTestCase(testsCase *analysis.TestCase, horusecConfigFilePath, runType string) {
  ...
}
```

Reply:
updated! :)